foodji_api/Application/Command/UpdateRecipeCommand.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Application.Dto;
+using AutoMapper;
+using Infra;
+using MediatR;
+using Domain.Recipes;
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace Application.Command;
+
+public class UpdateRecipeCommand : IRequest<string>
+{
+    private RecipeDto RecipeDto { get; }
+    
+    public UpdateRecipeCommand(RecipeDto recipeDto)
+    {
+        RecipeDto = recipeDto;
+    }
+    
+    private class Handler : IRequestHandler<UpdateRecipeCommand, string>
+    {
+        private readonly IFoodjiDbClient _client;
+        private readonly IMapper _mapper;
+
+        public Handler(IFoodjiDbClient client, IMapper mapper)
+        {
+            _client = client;
+            _mapper = mapper;
+        }
+        
+        public async Task<string> Handle(UpdateRecipeCommand request, CancellationToken cancellationToken)
+        {
+            var recipe = _mapper.Map<Recipe>(request.RecipeDto);
+            
+            var results = await _client.Recipes.FindAsync(
+                x => x.Id == new ObjectId(request.RecipeDto.Id),
+                cancellationToken: cancellationToken);
+
+            var recipeToUpdateList = results.ToList(cancellationToken);
+            
+            switch (recipeToUpdateList.Count)
+            {
+                case > 1:
+                    // TODO More specific exception to go along better exception handling in the API layer
+                    //      500 many with the same ID (bad news!)
+                    //      shouldn't happen, but no "FindOne" method to make that check for us
+                    throw new Exception($"{recipeToUpdateList.Count} recipes with the id {request.RecipeDto.Id} found");
+                case 0:
+                    return string.Empty;
+            }
+
+            var recipeToUpdate = recipeToUpdateList.First();
```

Comment:
Intéressant que tu ailles vers par défaut vers du LINQ pour touts tes accès à des listes. Je n'ai pas nécessairement de point à soulever contre, pour être bien franche, et peut-être même est-ce mieux par certains aspect. 

J'en profite pour soulever que LINQ fonctionne par `IQueryable` à l'interne, qui se comportent un peut différemment (point soulevé purement pour purpose of knowledge-sharing). Voir 

> The IQueryable<> variable has deferred execution, therefore internally, EF will only make the call to the database when the IQueryable<> is iterated over (i.e. when .MoveNext() is first called).

https://stackoverflow.com/a/57232381

IIRC, c'est pratique lorsque tu enchaînes les requêtes. Un `.Any` peut commencer à regarder aussitôt qu'il reçoit le premier résultat de la query précédente, je pense bien, et short-circuit avant que le reste de la requête précédente (un `.Select`mettons) n'ait à s'exécuter au complet. C'est notament pratique si tu veux travailler avec de grandes quantités de données et pas tout loader en mémoire d'un coup (ça reste un **L**anguage **In**tegrated ***Q**uery language, LINQ - query as is with databases).

Bref, juste un point intéressant qui peut valoir plus de lecture si jamais ça te chante, ou qui peut revenir en tête dans 2 ans quand tu fais face à un comportement niche étrange hahah - apologies if it's totally unasked for and unwarranted. #nerdMomentOver

Reply:
Je vais lire là dessus 😉 merci de la recommandation!

Je commence tranquillement pas vite à m'habituer avec le stock de .NET, le First me semblait pratique et faisait pas mal ce que j'avais de besoin, mais j'aurais pu utiliser le classique [0] as well sans trop de difficulté!

Reply:
Yep, ça fait parfaitement la job! Au plus, ça me surprend parce que ce n'est pas de quoi uqi est revenu souvent, mais honnêtement, c'est clean.